classes/local/store/swift/stream_wrapper.php

```diff
@@ -431,6 +431,33 @@ public function unlink($path) {
         }
     }
 
+    /**
+     * Rename object
+     *
+     * @param string $path
+     * @return boolean
+     */
+
+    public function rename($currentpath, $destinationpath) {
+        $currenturl = $this->parse_url($currentpath);
+        $destinationurl = $this->parse_url($destinationpath);
+
+        $options = ['name' => $destinationurl['path'], 'containerName' => $destinationurl['host']];
+        try {
+            $this->get_container($currenturl['host'])->getObject($currenturl['path'])->copy($options);
```

Comment:
I think this needs to be changed to `['destination' => $pathhere]` as per https://php-openstack-sdk.readthedocs.io/en/latest/services/object-store/v1/objects.html#copy-object
Otherwise I get a 412 precondition failed destination header required
With that in place I get 403 forbidden, this looks to be something specific to the openstack system I'm testing with

Reply:
Line 445 needs to be `$options = ['destination' => $destinationurl['host'] . '/' . $destinationurl['path']];`
That will correctly move the file.